independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/TypesTest.java

```diff
@@ -67,15 +70,41 @@ public void testGetTypeClosure() throws IOException {
         String producersName = "produce";
         MethodInfo producerMethod = producerClass.method(producersName);
         // Object is the sole type
-        Set<Type> producerMethodTypes = Types.getProducerMethodTypeClosure(producerMethod,
-                dummyDeployment);
-        assertEquals(1, producerMethodTypes.size());
+        try {
+            Types.getProducerMethodTypeClosure(producerMethod, dummyDeployment);
+            fail("Should have thrown definition exception. Wildcard is not a legal bean type!");
+        } catch (DefinitionException e) {
```

Comment:
The `try` block is ok but `Assertions.assertThrows()` is sometimes more readable...

Reply:
Fair enough, changed :)